pyanaconda/gnome_remote_destop.py

```diff
@@ -0,0 +1,183 @@
+            stdoutLog.critical("No openssl binary found, can't generate certificates for GNOME remote desktop. Aborting.")
+            util.ipmi_abort(scripts=self.anaconda.ksdata.scripts)
+            sys.exit(1)
+
+        # start by checking we have openssl available
+        if not os.path.exists(GRD_BINARY_PATH):
+            # we assume there that the main binary being present implies grdctl is there as well
+            stdoutLog.critical("GNOME remote desktop tooling is not available. Aborting.")
+            util.ipmi_abort(scripts=self.anaconda.ksdata.scripts)
+            sys.exit(1)
+
+    def _handle_rdp_certificates(self):
+        """Generate SSL certificate and use it for incoming RDP connection."""
+
+        # then create folder for the certs
+        os.makedirs(GRD_RDP_CERT_DIR)
+        # generate the certs
+        # TODO: check openssl binary is available ?
+        execWithCapture(OPENSSL_BINARY_PATH,
+                        ["req", "-new",
+                         "-newkey" "rsa:4096",
+                         "-days 720", "-nodes", "-x509",
+                         "-subj", "/C=DE/ST=NONE/L=NONE/O=GNOME/CN=localhost",
+                         "-out", GRD_RDP_CERT,
+                         "-keyout", GRD_RDP_CERT_KEY]
+                        )
+        # tell grd to use these certificates
+        execWithCapture("grdctl", ["--headless", "rdp", "set-tls-cert", GRD_RDP_CERT])
+        execWithCapture("grdctl", ["--headless", "rdp", "set-tls-key", GRD_RDP_CERT_KEY])
+
+    def _set_rdp_username_and_password(self):
+        """Set the RDP username and password."""
+        execWithCapture("grdctl", ["--headless", "rdp", "set-credentials", self.rdp_username, self.rdp_password])
+        # also actually tell GRD that we (obviously) want to use RDP
+        execWithCapture("grdctl", ["--headless", "rdp", "enable"])
+
+    def _find_network_address(self):
+        """Find machine IP address, so we can show it to the user."""
+
+        # Network may be slow. Try for 5 seconds
+        tries = 5
+        while tries:
+            self.ip = network.get_first_ip_address()
+            if self.ip:
+                break
+            time.sleep(1)
+            tries -= 1
+
+        if not self.ip:
+            return
+
+        if self.ip.find(':') != -1:
+            ipstr = "[%s]" % (self.ip,)
+        else:
+            ipstr = self.ip
+
+        try:
+            hinfo = socket.gethostbyaddr(self.ip)
+            if len(hinfo) == 3:
+                # Consider as coming from a valid DNS record only if single IP is returned
+                if len(hinfo[2]) == 1:
+                    self.name = hinfo[0]
+        except socket.herror as e:
+            log.debug("Exception caught trying to get host name of %s: %s", ipstr, e)
+
+    def _start_grd_process(self):
+        """Start the GNOME remote desktop process."""
+        try:
+            execWithCapture(GRD_BINARY_PATH, [])
```

Comment:
I think that you need to add the `--headless` param here:
https://gitlab.gnome.org/GNOME/gnome-remote-desktop/-/blob/master/data/gnome-remote-desktop-headless.service.in?ref_type=heads

Also, `execWithCapture` will block until the app starts. I believe you need to run it in the background.

Reply:
Yep, running with `startProgram()` now, that does not wait for it to terminate. Still need to change the shutdown logic to account for that probably. :)